pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -122,7 +122,7 @@ func TestDeploymentController_reconcileNewRC(t *testing.T) {
 	}
 }
 
-func TestDeploymentController_reconcileOldRCs(t *testing.T) {
+func TestDeploymentController_scaleDownOldRCsForRollingUpdate(t *testing.T) {
```

Comment:
Maybe keep the test for `reconcileOldRCs`, and add tests for `scaleDownOldRCsForRollingUpdate` and `cleanupUnhealthyReplicas`?